.buildkite/pipeline.yml

```diff
@@ -0,0 +1,73 @@
+# Nodes with values to reuse in the pipeline.
+common_params:
+  plugins: &common_plugins
+  - &bash_cache automattic/bash-cache#v1.3.1: ~
+  # Common environment values to use with the `env` key.
+  env: &common_env
+    IMAGE_ID: xcode-12.5.1
+
+# This is the default pipeline – it will build and test the app
+steps:
+  ######################
+  # Build and Test iOS
+  ######################
+  - label: "🧪 Build and Test iOS"
+    key: "test_ios"
+    command: "build_and_test_pod 'ios test'"
+    env: *common_env
+    plugins: *common_plugins
+    artifact_paths: ".build/logs/*.log"
+
+  ######################
+  # Build and Test macOS
+  ######################
+  - label: "🧪 Build and Test macOS"
+    key: "test_macos"
+    command: "build_and_test_pod 'mac test'"
```

Comment:
Catching up on Buildkite and I'm wondering where do those `build_and_test_pod`, `lint_pod`, etc… commands come from, aka where are they defined?

Are those built-in commands that BuildKite provide to us? Are those scripts of our own that are installed in the `$PATH` of all our BuildKite mac agents?

And given that `build_and_test_pod` seems to take as a parameter just the name of a lane in the `Fastfile`, I'm curious of the benefit of having `build_and_test_pod 'mac test'` as opposed to `bundle exec fastlane mac test` directly for clarity?

Reply:
These functions exist as part of this plugin: https://github.com/Automattic/bash-cache-buildkite-plugin

It's what's referred to as [a library plugin](https://github.com/buildkite-plugins/library-example-buildkite-plugin) in Buildkite parlance. They're exposed on the `$PATH`, as you say. 

The [`build_and_test_pod`](https://github.com/Automattic/bash-cache-buildkite-plugin/blob/trunk/bin/build_and_test_pod) call is a high-level action that installs gems, pods (if needed), and kicks off the specified lane – hopefully seeing the implementation helps clarify the intent a bit?

Reply:
Ahhh, I looked at the wrong repo (I looked at our private `buildkite-ci` one instead) 🤦 

Tbh I didn't even think of looking in that `bash-cache` plugin because the name of that plugin didn't make me think it would provide those helper commands – I thought it was only about caching (and didn't really wondered what things it'd actually cache tbh), not about providing higher level helper commands.

I now understand that this started as a plugin focused on caching (around gems, Pods/, etc), but it seems it has since grown to provide way more custom commands, most of those sure ending up calling something that involves cache (like `build_and_test_pod` calling `install_gems` internally), but that feels like an implementation detail to me 😅  . And even some of those don't involve caching at all (e.g. `slack_notify_pod_published`)

And it seems like there are now more helper commands there that are _not_ directly related to cache management, than commands that do… So I'm wondering if it wouldn't make sense to rename that library plugin to something like `bash-commands-buildkite-plugin` or something like that? (Would definitively have avoided me the confusion and missing those at least 😅 )